File: ulohy/31-transpozice/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let w;
let h;

const table = [];

let rowsLoaded = 0;
let colsLoaded = 0;

rl.on("line", (line) => {
    if (w === undefined) {
        w = parseInt(line);
        return;
    }

    if (h === undefined) {
        h = parseInt(line);
        for (let i = 0; i < h; i++) {
            table.push([]);
        }

        if (h < 1 || w < 1) {
            rl.close();
        }
        return;
    }

    if (line === "") {
        return;
    }

    table[rowsLoaded].push(line);

    colsLoaded++;
    if (colsLoaded >= w) {
        rowsLoaded++;
        colsLoaded = 0;
    }

    if (rowsLoaded < h) {
        return;
    }

    for (let col = 0; col < w; col++) {
        for (let row = 0; row < h; row++) {
            process.stdout.write(table[row][col]);
        }
        process.stdout.write("\n");
    }

    rl.close();
});

File: ulohy/31-transpozice/reseni/main.cpp
#include <iostream>
#include <vector>

int main() {
    int w, h;
    std::cin >> w >> h;

    if (w < 1 || h < 1) {
        return 0;
    }

    std::vector<std::vector<char>> table(h, std::vector<char>(w));

    for (int row = 0; row < h; row++) {
        for (int col = 0; col < w; col++) {
            std::cin >> table[row][col];
        }
    }

    for (int col = 0; col < w; col++) {
        for (int row = 0; row < h; row++) {
            std::cout << table[row][col];
        }
        std::cout << std::endl;
    }

    return 0;
}
